packages/clay-alert/src/ClayAlert.tsx

```diff
@@ -0,0 +1,177 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import Icon from '@clayui/icon';
+
+const useAutoClose = (autoClose?: boolean | number, onClose = () => {}) => {
+	const timer = React.useRef<number | undefined>(undefined);
+
+	let cancelTimer = () => {};
+	let startTimer = () => {};
+
+	if (autoClose) {
+		cancelTimer = () => {
+			if (timer.current) {
+				clearTimeout(timer.current);
+
+				timer.current = undefined;
+			}
+		};
+
+		startTimer = () => {
+			timer.current = setTimeout(
+				onClose,
+				autoClose === true ? 8000 : autoClose
+			);
+		};
+	}
+
+	React.useEffect(() => {
```

Comment:
I think we can still call `cancelTimer` on the return of `useEffect` to clear the timer, when the component is dismounting and if it has not finished the time.

Reply:
Yeah I am doing that inside of the `if (autoClose && onClose)` condition. Are you thinking we should move it outside of that condition?

Reply:
Sorry Bryce I had confused about it, that's correct. 😅